sqlserver/datadog_checks/sqlserver/activity.py

```diff
@@ -67,10 +74,15 @@
 FROM sys.dm_exec_sessions sess
     INNER JOIN sys.dm_exec_connections c
         ON sess.session_id = c.session_id
-    INNER JOIN sys.dm_exec_requests req
+    FULL OUTER JOIN sys.dm_exec_requests req
         ON c.connection_id = req.connection_id
-    CROSS APPLY sys.dm_exec_sql_text(req.sql_handle) qt
-WHERE sess.session_id != @@spid AND sess.status != 'sleeping'
+    OUTER APPLY sys.dm_exec_sql_text(req.sql_handle) qt
+    OUTER APPLY sys.dm_exec_sql_text(c.most_recent_sql_handle) lqt
+WHERE sess.session_id != @@spid
+    AND (sess.status != 'sleeping'
+         OR sess.session_id IN (SELECT blocking_session_id FROM cteBlocking)
```

Comment:
If this anti-join causes a problem, a potential solution is to do the 2nd run it only in case when we find some idle blockers. The heuristic here is, that idle blockers are an exceptional event.